tests/api/api/test_projects.py

```diff
@@ -970,15 +971,80 @@ def test_delete_project_not_found_in_leader(
     assert response.status_code == HTTPStatus.CREATED.value
     _assert_project_response(project, response)
 
-    response = unversioned_client.delete(
-        f"{delete_api_version}/projects/{project.metadata.name}",
-    )
-    assert response.status_code == HTTPStatus.ACCEPTED.value
+    with unittest.mock.patch.object(
+        mock_project_follower_iguazio_client,
+        "delete_project",
+        side_effect=mlrun.errors.MLRunNotFoundError("Project not found"),
+    ):
+        response = unversioned_client.delete(
+            f"{delete_api_version}/projects/{project.metadata.name}",
+        )
+        assert response.status_code == HTTPStatus.ACCEPTED.value
 
-    response = unversioned_client.get(
-        f"v1/projects/{project.metadata.name}",
+        response = unversioned_client.get(
+            f"v1/projects/{project.metadata.name}",
+        )
+        assert response.status_code == HTTPStatus.NOT_FOUND.value
+
+
+# Test should not run more than a few seconds because we test that if the background task fails,
+# the wrapper task fails fast
+@pytest.mark.timeout(10)
+@pytest.mark.parametrize(
+    "delete_api_version",
+    [
+        "v1",
+        "v2",
+    ],
+)
+def test_delete_project_fail_fast(
+    unversioned_client: TestClient,
+    mock_project_follower_iguazio_client,
+    delete_api_version: str,
+) -> None:
+    # Set the igz version for the project leader mock
+    # We only test igz version < 3.5.5 flow because from 3.5.5 iguazio waits for the inner background task to
+    # finish so the wrapper task does not wait for the inner task
```

Comment:
isn't the fail fast what you are testing? so it happens only for 3.5.5, no?

Reply:
We only wait for the project deletion on iguazio < 3.5.5.
From iguazio 3.5.5 zebo waits for the inner bg task so the wrapper task waits on the zebo job and we know that if the zebo job succeeded/failed then the inner bg task finished. Therfore there is nothing to test on igz 3.5.5 because the flow doesn't reach the wait for deletion part.